.github/workflows/build.yml

```diff
@@ -6,16 +6,15 @@ on:
       - main
       - release/*
   pull_request:
-  merge_group:
   workflow_dispatch:
 
-permissions:
-  contents: read
-
 concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
   cancel-in-progress: true
 
+permissions:
+  contents: read
+
```

Comment:
do you mind reverting this change? I think permissions are more important than concurrency, so prefer it to be at the top